kpatch-build/kpatch-build

```diff
@@ -464,19 +464,24 @@ cp "$PATCHFILE" "$APPLIEDPATCHFILE" || die
 cp -LR "$DATADIR/patch" "$TEMPDIR" || die
 export KCFLAGS="-I$DATADIR/patch -ffunction-sections -fdata-sections"
 
-echo "Reading special section data"
-SPECIAL_VARS=$(readelf -wi "$VMLINUX" |
-	       gawk --non-decimal-data '
-	       BEGIN { a = b = p = e = 0 }
-	       a == 0 && /DW_AT_name.* alt_instr\s*$/ {a = 1; next}
-	       b == 0 && /DW_AT_name.* bug_entry\s*$/ {b = 1; next}
-	       p == 0 && /DW_AT_name.* paravirt_patch_site\s*$/ {p = 1; next}
-	       e == 0 && /DW_AT_name.* exception_table_entry\s*$/ {e = 1; next}
-	       a == 1 {printf("export ALT_STRUCT_SIZE=%d\n", $4); a = 2}
-	       b == 1 {printf("export BUG_STRUCT_SIZE=%d\n", $4); b = 2}
-	       p == 1 {printf("export PARA_STRUCT_SIZE=%d\n", $4); p = 2}
-	       e == 1 {printf("export EX_STRUCT_SIZE=%d\n", $4); e = 2}
-	       a == 2 && b == 2 && p == 2 && e == 2 {exit}')
+if grep "CONFIG_DEBUG_KERNEL=y" "$OBJDIR/.config" > /dev/null; then
+	# The kernel supports debug infos.
+	echo "Reading special section data"
+	SPECIAL_VARS=$(readelf -wi "$VMLINUX" |
+		       gawk --non-decimal-data '
+		       BEGIN { a = b = p = e = 0 }
+		       a == 0 && /DW_AT_name.* alt_instr\s*$/ {a = 1; next}
+		       b == 0 && /DW_AT_name.* bug_entry\s*$/ {b = 1; next}
+		       p == 0 && /DW_AT_name.* paravirt_patch_site\s*$/ {p = 1; next}
+		       e == 0 && /DW_AT_name.* exception_table_entry\s*$/ {e = 1; next}
+		       a == 1 {printf("export ALT_STRUCT_SIZE=%d\n", $4); a = 2}
+		       b == 1 {printf("export BUG_STRUCT_SIZE=%d\n", $4); b = 2}
+		       p == 1 {printf("export PARA_STRUCT_SIZE=%d\n", $4); p = 2}
+		       e == 1 {printf("export EX_STRUCT_SIZE=%d\n", $4); e = 2}
+		       a == 2 && b == 2 && p == 2 && e == 2 {exit}')
+else
+	die "no support for debug infos"
```

Comment:
This error message should be more specific, like "kernel doesn't have 'CONFIG_DEBUG_KERNEL' enabled".